### DISCUSSION TITLE
Infix named args

### DISCUSSION THREAD

**Discussion by som-snytt** [scala/scala3#19072] [General Discussion]
This is weird:
```
42 + (x = 1)
```
Is no one opinionated about infix? I mean to ask, isn't anyone?

**Comment by nicolasstucki** [comment]
At first I parsed this as an assignment in expression position.

**Reply by som-snytt** [reply]
That's all I'm saying, baby.

**Comment by som-snytt** [comment]
Thanks for the feedback, which makes me feel supported and less lonely.

via discord, an authoritative agree and an authoritative poop emoji:

![image](https://github.com/lampepfl/dotty/assets/369425/61001081-359e-422a-aaaa-6914e9e8c27d)

**Comment by som-snytt** [comment]
Since a discussion is not an issue, it doesn't show a mention of https://github.com/scala/scala3/pull/21565.

What was so terrible about this syntax again? It just looks funny; but many legal things look funny, that is what democracy and free speech is all about.

An "argument expression" can be an assignment syntactically, which may be a named arg in parens. What's the problem?

The reason to deter "multi-arg infix" was that the args "look like a tuple"; similarly, named args look like a named tuple.

Anyone can choose to lint that syntax if they wish, but it's not scalac's job.

**Comment by som-snytt** [comment]
For reference
https://github.com/scala/scala3/commit/5d5a9e6ff17a5d1980653e6fcb622dc5a2e4d082
